modules/account-lib/test/unit/coin/algo/transactionBuilder/keyRegistrationBuilder.ts

```diff
@@ -0,0 +1,286 @@
+import { coins } from '@bitgo/statics';
+import algosdk from 'algosdk';
+import should from 'should';
+import sinon, { assert } from 'sinon';
+import { Transaction } from '../../../../../src/coin/algo';
+import { KeyRegistrationBuilder } from '../../../../../src/coin/algo/keyRegistrationBuilder';
+import utils from '../../../../../src/coin/algo/utils';
+
+import * as AlgoResources from '../../../../resources/algo';
+
+class StubTransactionBuilder extends KeyRegistrationBuilder {
+  getTransaction(): Transaction {
+    return this._transaction;
+  }
+}
+
+describe('Algo KeyRegistration Builder', () => {
+  let builder: StubTransactionBuilder;
+  
+  const sender = AlgoResources.accounts.account1;
+  const { rawTx } = AlgoResources
+
+
+  beforeEach(() => {
+    const config = coins.get('algo');
+    builder = new StubTransactionBuilder(config)
+  });
+
+  describe('setter validation', () => {
+    it('should validate voteKey, is set and is a valid string', () => {
+      const spy = sinon.spy(builder, 'voteKey');
+      should.doesNotThrow(() => builder.voteKey(sender.voteKey));
+      assert.calledOnce(spy);
+    });
+
+    it('should validate selection key, is set and is a valid string', () => {
+      const spy = sinon.spy(builder, 'selectionKey');
+      should.doesNotThrow(() => builder.selectionKey(sender.selectionKey));
+      assert.calledOnce(spy);
+    });
+
+    it('should validate vote First key is gt than 0', () => {
+      const spy = sinon.spy(builder, 'validateValue');
```

Comment:
this spy is good because it verifies validate value was called where we don't call the method directly.